network/filters.go

```diff
@@ -41,3 +41,44 @@ func isFailBefore(rep reputationManager, cutoff time.Time) func(string) bool {
 		return rep.Fail(address).Before(cutoff)
 	}
 }
+
+//MsgType enum.
+type MsgType uint16
+
+const (
+	connected    MsgType = 1
+	disconnected MsgType = 2
+	received     MsgType = 3
+)
+
+//MsgFilter can be used to filter out message types.
+func MsgFilter(msgType MsgType, addresses ...string) func(interface{}) bool {
+	return func(msg interface{}) bool {
+		switch message := msg.(type) {
+		case *Connected:
+			return msgType == connected && len(addresses) == 0 || contains(addresses, message.Address)
```

Comment:
I think this probably evaluates wrong. Either way, it's probably better to use parantheses to clarify.
It's always preferred in Go, by the way, to write more code but be more explicit. So I would probably write this in several lines.